### PR TITLE
Refactor: `FSMBehaviourTestToolSetup`

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

Base class for testing test tools

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: packages/valory/skills/abstract_round_abci/tests/test_tools/test_common.py [hunk 1]
```diff
@@ -175,14 +123,14 @@ class TestBaseRandomnessBehaviourTestRunning(BaseRandomnessBehaviourTest):
...
 
-    _test_cls = BaseSelectKeeperBehaviourTest
+    test_cls: Type[BaseSelectKeeperBehaviourTest] = BaseSelectKeeperBehaviourTest
 
-    def run_test(self, test_instance: BaseSelectKeeperBehaviourTest) -> None:
-        """Run test"""
-        test_instance.test_select_keeper_preexisting_keeper()
+    def set_select_keeper_behaviour_class(self) -> None:
+        """Set select_keeper_behaviour_class"""
+        self.test_cls.select_keeper_behaviour_class = DummyKeeperSelectionBehaviour  # type: ignore
```
Comment (DavidMinarsch) [inline packages/valory/skills/abstract_round_abci/tests/test_tools/test_common.py:130]: why type ignore?

File: packages/valory/skills/abstract_round_abci/tests/test_tools/base.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    def teardown(self) -> None:
+        """Teardown test"""
+        self.test_cls.teardown_class()
+
+    def set_path_to_skill(self, path_to_skill: Path = PATH_TO_SKILL) -> None:
+        """Set path_to_skill"""
+        self.test_cls.path_to_skill = path_to_skill
+
+    def setup_test_cls(self, **kwargs: Any) -> FSMBehaviourBaseCase:
```
Comment (DavidMinarsch) [inline packages/valory/skills/abstract_round_abci/tests/test_tools/base.py:73]: Where is this being called?

Comment (Karrenbelt) [reply packages/valory/skills/abstract_round_abci/tests/test_tools/base.py:73]:
```
- abstract_round_abci/tests/test_tools/test_base.py
- abstract_round_abci/tests/test_tools/test_common.py
- abstract_round_abci/tests/test_tools/test_integration.py
```

File: packages/valory/skills/abstract_round_abci/tests/test_tools/test_common.py (hunk 1 above)
Comment (Karrenbelt) [reply packages/valory/skills/abstract_round_abci/tests/test_tools/test_common.py:130]: got this on CI, tho not local
```
mypy run-test: commands[1] | mypy autonomy packages/valory plugins/aea-test-autonomy/aea_test_autonomy tests scripts --disallow-untyped-defs
packages/valory/skills/abstract_round_abci/tests/test_tools/test_common.py:67: error: Can only assign concrete classes to a variable of type "Type[BaseBehaviour]"
packages/valory/skills/abstract_round_abci/tests/test_tools/test_common.py:130: error: Can only assign concrete classes to a variable of type "Type[BaseBehaviour]"
Found 2 errors in 1 file (checked 439 source files)
```

Comment (DavidMinarsch) [reply packages/valory/skills/abstract_round_abci/tests/test_tools/test_common.py:130]: Makes sense. It’s a known issue with this type

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**